### PR TITLE
JIT - handle multiple expressions (as returned from CSE)

### PR DESCRIPTION
**Pull request by mdewing**
The `expr` argument to llvm_callable can either be a single expression, or a structure that matches the return value from `cse`.

The expected structure is a tuple or list with two items - first item is a list of (Symbol, expr) pairs, and the second item is a list with one item - the final expression to be returned.
(The cse module can handle multiple expressions, and in that case will put multiple items in that second list.  This PR will only process and return the value from the first expression on that list)

This code also assumes the list of (Symbol, expr) is topologically sorted.

Example usage

``` python
from sympy import *
import sympy.printing.llvmjitcode as jit
from sympy.abc import a,x,y,z

e = x*x + y*y  + exp(-x*x- y*y)
f = x*x + exp(-x*x- y*y)
print('e = ',e)
e2 = cse(e)
print('e2 = ',e2)

j1 = jit.llvm_callable([x,y],e)
print("original expression = ",j1(1.0, 2.0))

j2 = jit.llvm_callable([x,y],e2)
print("after cse = ",j2(1.0, 2.0))

```

### CODE REVIEW

**Comment by asmeurer** [comment]
> This PR will only process and return the value from the first expression on that list

It might be better to just raise a NotImplementedError. Only processing the first item doesn't sound like expected behavior.

**Comment by markdewing** [comment]
Here's an example using 6-D integration to approximate the ground state energy of a helium atom.  It also shows the use of CSE to improve the performance of the callback (and the integration) - 195 seconds vs. 140 seconds with CSE.

https://github.com/markdewing/next_steps_in_programming/blob/master/examples/integration/Helium%20atom.ipynb

**Comment by moorepants** [comment]
Great example! Feel free to submit that for our examples directory.

File: sympy/printing/tests/test_llvmjit.py [hunk 1]
```diff
@@ -44,10 +45,138 @@ def test_func():
...
+    assert isclose(jit_res, res)
+
+
+def test_cse_multiple():
+    e1 = a*a
+    e2 = a*a + b*b
+    e3 = sympy.cse([e1, e2])
+    try:
+        g.llvm_callable([a, b], e3)
+    except NotImplementedError:
```
Comment (asmeurer) [inline sympy/printing/tests/test_llvmjit.py]: You should use the `raises` function here.

**Comment by markdewing** [comment]
Ping @asmeurer @moorepants 
Could this PR get merged?

**Pull request merged**